plutus-core/exe/Main.hs

```diff
@@ -686,23 +685,23 @@ runEval :: EvalOptions -> IO ()
 runEval (EvalOptions language inp ifmt evalMode printMode timingMode) =
     case language of
 
-      TypedPLC -> do
-        TypedProgram prog <- getProgram TypedPLC ifmt inp
-        let evaluate =
-                case evalMode of
-                  CK  -> PLC.unsafeEvaluateCk  PLC.defBuiltinsRuntime
-                  CEK -> PLC.unsafeEvaluateCek PLC.defBuiltinsRuntime
-            body = void . PLC.toTerm $ prog
-            !_ = rnf body
-        -- Force evaluation of body to ensure that we're not timing parsing/deserialisation.
-        -- The parser apparently returns a fully-evaluated AST, but let's be on the safe side.
-        case timingMode of
-          NoTiming -> evaluate body & handleResult
-          Timing n -> timeEval n evaluate body >>= handleTimingResults
+      TypedPLC ->
+        case evalMode of
+            CEK -> hPutStrLn stderr "There is no CEK machine for Typed Plutus Core" >> exitFailure
```

Comment:
What happens now if you say `plc evaluate -i file.plc --typed`?  I presume you get the error message, so you'll always have to type  `plc evaluate -i file.plc --typed -mCK`.  I guess that's OK though.  We could do away with the `-m` option and make it choose depending on whether we're executing something typed or untyped, but that's probably a bad idea sine we may want to add other evaluators (eg HOAS) in future. It's probably better just to require the user to always specify `-mCK` for typed PLC so that they remember that they're using a CK machine, whose memory consumption etc. may differ from that of the CEK machine.

Reply:
Yeah, seems fine for now at least.

Reply:
Ok, I'll leave it as is.